### PR TITLE
Ensure that the staging badge displays when submitted against a staging project

### PR DESCRIPTION
**Pull request by hellcp-work**
While reviewing another PR, @hennevogel noticed that the staging badge displays when a review is created for a staging project that's not necessarily a staging project of the current target project. This should fix that

### CODE REVIEW

**Comment by danidoni** [comment]
I believe this change needs to be backed by specs

File: src/api/app/controllers/webui/request_controller.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def show
                                           diffs: true, action_id: action_id.to_i, cacheonly: 1).first
       @active_action = @bs_request.bs_request_actions.find(action_id)
 
-      @request_reviews = @bs_request.reviews.for_non_staging_projects
+      @request_reviews = @bs_request.reviews.for_non_staging_projects(Project.find_by_name(@bs_request.target_project_name))
```
Comment (dmarcoux) [inline src/api/app/controllers/webui/request_controller.rb:31]: This is now repeating the same database query which happens a few lines below on line 45. Please, avoid this by storing the result in a variable.

File: src/api/app/models/review.rb [hunk 2]
```diff
@@ -60,10 +60,10 @@ class NotFoundError < APIError
   scope :opened, -> { where(state: :new) }
   scope :accepted, -> { where(state: :accepted) }
   scope :declined, -> { where(state: :declined) }
-  scope :for_staging_projects, -> { includes(:project).where.not(projects: { staging_workflow_id: nil }) }
-  scope :for_non_staging_projects, -> { includes(:project).where(projects: { staging_workflow_id: nil }) }
+  scope :for_staging_projects, ->(project) { includes(:project).where(projects: { staging_workflow: Staging::Workflow.find_by(project:) }).where.not(projects: { staging_workflow_id: nil }) }
```
Comment (dmarcoux) [inline src/api/app/models/review.rb:63]: The value is missing for the `project:` key in the `Staging::Workflow#find_by(...)` call.

File: src/api/spec/features/webui/requests_spec.rb [hunk 3]
```diff
@@ -374,4 +374,42 @@
       expect(page).not_to have_text('Project Maintainers')
     end
   end
+
+  describe 'a request submitted against a staging project' do
+    let!(:workflow) do
+      create(:staging_workflow_with_sources)
+      Staging::Workflow.last
+    end
+    let!(:staging_project) { workflow.staging_projects.first }
```
Comment (danidoni) [inline src/api/spec/features/webui/requests_spec.rb:383]:
```suggestion
    let(:staging_project) { workflow.staging_projects.first }
```
You don't really need the bang here, as you can just lazy load it when needed

**Comment by dmarcoux** [comment]
Do you plan on addressing the CodeClimate issues?

File: src/api/spec/features/webui/requests_spec.rb [hunk 4]
```diff
@@ -374,4 +374,54 @@
...
+      create(:staging_workflow, project: create(:project, name: 'home:titan:stage', maintainer: submitter))
+      Staging::Workflow.last
+    end
+    let(:staging_project) { workflow.staging_projects.first }
+
+    before do
+      Flipper.enable(:request_show_redesign)
+    end
+
+    it 'does not set badge for sr' do
```
Comment (dmarcoux) [inline src/api/spec/features/webui/requests_spec.rb:389]: I assume `sr` stands for `submit request`? Please write the complete words.

File: src/api/spec/features/webui/requests_spec.rb [hunk 5]
```diff
@@ -374,4 +374,54 @@
...
+    end
+    let(:staging_project) { workflow.staging_projects.first }
+    let(:staging_request) { create(:delete_bs_request, target_project: target_project, creator: submitter) }
+    let(:staging_user) { User.find_by(login: staging_request.creator) }
+
+    before do
+      Flipper.enable(:request_show_redesign)
+    end
+
+    it 'sets badge for sr' do
```
Comment (dmarcoux) [inline src/api/spec/features/webui/requests_spec.rb:415]: Same as above, let's write the complete words.

**Review by danidoni — DISMISSED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**